lib/vector-core/src/event/metric.rs

```diff
@@ -1237,18 +1237,19 @@ fn write_word(fmt: &mut Formatter<'_>, word: &str) -> Result<(), fmt::Error> {
     }
 }
 
-pub fn samples_to_buckets(samples: &[Sample], buckets: &[f64]) -> (Vec<Bucket>, u32, f64) {
+fn samples_to_buckets(samples: &[Sample], buckets: &[f64]) -> (Vec<Bucket>, u32, f64) {
     let mut counts = vec![0; buckets.len()];
     let mut sum = 0.0;
     let mut count = 0;
     for sample in samples {
-        buckets
+        if let Some((i, _)) = buckets
             .iter()
             .enumerate()
             .skip_while(|&(_, b)| *b < sample.value)
```

Comment:
As an aside, and not something we need to address here: I think what end up being more generically compatible, in terms of ecosystem/interoperation, would be having buckets be "left inclusive/right exclusive".  For example, if we had two buckets -- upper bounds of 1.0 and 2.0, respectively -- we would describe the second bucket as encompassing all values from (and including) 1.0 up to (but not including) 2.0.

This seems to be the norm in a few numercial/statistical libraries and products/services I sampled:  NumPy, `ndarray_stats`, MATLAB, Circonus, and others.

The trouble with this is that Prometheus, in particular, expects things to be left exclusive/right inclusive... and we can't transform between the two schemes without the original samples.  Something for another day, though.